Directory.Build.props

```diff
@@ -68,7 +68,8 @@
         - src/tasks/MobileBuildTasks/Apple/AppleProject.cs
         - https://github.com/dotnet/sdk repo > src/Installer/redist-installer/targets/GeneratePKG.targets
      -->
-    <AndroidApiLevelMin>21</AndroidApiLevelMin>
+    <AndroidApiLevelMin Condition="'$(RuntimeFlavor)' != 'CoreCLR'">21</AndroidApiLevelMin>
+    <AndroidApiLevelMin Condition="'$(RuntimeFlavor)' == 'CoreCLR'">29</AndroidApiLevelMin>
```

Comment:
What breaks when AndroidApiLevelMin is set to 21 (same as Mono)? Can we fix or workaround those breaks instead?

This change is problematic for VMR as @jkoritzinsky said. I would be best to avoid introducing VMR regressions as part of the Android CoreCLR. VMR is higher priority project compared to Android CoreCLR.

Reply:
From the native part of the runtime perspective, the only thing that breaks is the TLS access helpers: GetThreadStaticsVariableOffset, GetTLSResolverAddress. Older Android level uses so called emulated TLS. I have a prototype implementation of these helpers for it that compile and match what C++ would generate. But we cannot have both in the same binary, as the :tlsdesc: relocation type would not work / compile on the older NDK.

Reply:
There used to be other problem, the missing glob function in the older NDKs, but I've removed usage of glob recently.

Reply:
Fwiw, I viewed setting 29 as temporary. We should be able to move forward and revisit when @janvorli's changes are ready.

Reply:
> GetThreadStaticsVariableOffset, GetTLSResolverAddress.

These are optional perf optimizations. You can just make `CanJITOptimizeTLSAccess` return false on Android to disable these optional perf optimizations for now.